### PR TITLE
Remove obsolete code

### PR DESCRIPTION
**Pull request by ElizabethOkerio**
### Issues

*This pull request fixes #xxx.*

### Description

This PR removes obsolete code from the code base. I've removed methods, properties, types and interfaces that were marked as obsolete. 

### Checklist (Uncheck if it is not completed)

- [ ] *Test cases added*
- [ ] *Build and test with one-click build and test script passed*

### Additional work necessary

*If documentation update is needed, please add "Docs Needed" label to the issue and provide details about the required document change in the issue.*

### CODE REVIEW

File: src/Microsoft.OData.Core/Microsoft.OData.Core.csproj [hunk 1]
```diff
@@ -92,4 +92,16 @@
     <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Shipped.txt" />
     <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Unshipped.txt" />
   </ItemGroup>
+
+  <ItemGroup>
```
Comment (habbes) [inline src/Microsoft.OData.Core/Microsoft.OData.Core.csproj:96]: Do we need this change?

File: test/FunctionalTests/Framework/Workspaces/APICallLog.cs [hunk 2]
```diff
@@ -546,23 +539,6 @@ public void ChangeInterceptor<T>(DataService<T> service, string methodName, obje
     }
     #endregion
 
-    #region IExpandProvider
-#pragma warning disable 618 // Disable "obsolete" warning for the IExpandProvider interface. Used for backwards compatibilty.
-    public class IExpandProviderCallLog : InterfaceCallLog<IExpandProvider>
```
Comment (habbes) [inline test/FunctionalTests/Framework/Workspaces/APICallLog.cs:551]: How come this didn't have the `Obsolete` attribute. Are you sure it's obsolete?

Comment (ElizabethOkerio) [reply test/FunctionalTests/Framework/Workspaces/APICallLog.cs:551]: The `IExpandProvider` that it implements is the one that's obsolete.

File: test/FunctionalTests/Tests/DataServices/UnitTests/ServerUnitTests/Stubs/DataServiceProvider/DSPExpandProvider.cs [hunk 3]
```diff
@@ -14,7 +14,7 @@ namespace AstoriaUnitTests.Stubs.DataServiceProvider
     using System.Linq;
 
 
-    public class DSPExpandProvider : IExpandProvider
+    public class DSPExpandProvider
```
[the remark this answers is not in the document]
Comment (ElizabethOkerio) [reply test/FunctionalTests/Tests/DataServices/UnitTests/ServerUnitTests/Stubs/DataServiceProvider/DSPExpandProvider.cs:17]: It has other methods and properties other than the ones from the removed interface.

File: src/Microsoft.OData.Core/Microsoft.OData.Core.csproj [hunk 4]
```diff
@@ -92,4 +92,15 @@
     <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Shipped.txt" />
     <AdditionalFiles Include="PublicAPI/$(TargetFramework)/PublicAPI.Unshipped.txt" />
   </ItemGroup>
+  <ItemGroup>
+    <Service Include="{508349b6-6b84-4df5-91f0-309beebad82d}" />
```
Comment (gathogojr) [inline src/Microsoft.OData.Core/Microsoft.OData.Core.csproj:96]: I have the same question Clement had. You might had responded to him directly but I think it'd have helped to add the response here

Comment (ElizabethOkerio) [reply src/Microsoft.OData.Core/Microsoft.OData.Core.csproj]: I've removed this. Must have been added when I ran the .tt files.

**Review by habbes — APPROVED** [review]

**Pull request merged**